x-pack/plugins/search_inference_endpoints/public/components/all_inference_endpoints/render_table_columns/render_deployment_status/deployment_status.tsx

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiHealth } from '@elastic/eui';
+import { DeploymentStatusEnum } from '../../types';
+
+interface DeploymentStatusProps {
+  status: DeploymentStatusEnum;
+}
+
+export const DeploymentStatus: React.FC<DeploymentStatusProps> = ({ status }) => {
+  if (status === DeploymentStatusEnum.notApplicable) {
+    return null;
+  }
+
+  let statusColor: string;
+
+  switch (status) {
+    case DeploymentStatusEnum.deployed:
+      statusColor = 'success';
+      break;
+    case DeploymentStatusEnum.notDeployed:
+      statusColor = 'warning';
+      break;
+    default:
+      statusColor = 'danger';
+  }
+
+  return <EuiHealth data-test-subj={`table-column-deployment-${status}`} color={statusColor} />;
+};
```

Comment:
This probably should just be an `EuiIcon`. 
If statusColor === 'success' then we will show `iconType="dot"` with `color="success"`
If statusColor === 'danger' then we will show `iconType="warning"` with `color="danger"`

We'll want to review the [a11y guidelines ](https://eui.elastic.co/#/display/icons)here too if we do an icon: "For better accessibility it's always recommended to give a descriptive title based on the icon use."

Seems like `.notDeployed` would apply to the `danger` color, but maybe I'm missing the underlying deployment status.  (Assuming `warning` would be that it is actively deploying, or waiting)

Lastly, we should include a tooltip for this.

```jsx
<EuiToolTip
  position="top"
  content={"Model is deployed" | "Model is deploying" | "Model could not be deployed" } // for example
> 
... EuiIcon...
</EuiToolTip>
```

Reply:
Updated